simplicity-sys/src/bitstream.rs

```diff
@@ -0,0 +1,44 @@
+// Simplicity Bindings
+// Written in 2022 by
+//   Christian Lewe <clewe@blockstream.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+use libc::{c_char, c_int, c_uchar, FILE};
+
+/// Stream of bits.
+///
+/// Corresponds to `bitstream` in `depend/simplicity/bitstream.h`.
+#[repr(C)]
+pub struct Bitstream {
+    file: *mut FILE,
+    available: c_int,
+    byte: c_uchar,
+}
+
+impl From<&[u8]> for Bitstream {
+    fn from(bytes: &[u8]) -> Self {
```

Comment:
In 8e45d76bc0e528aed678978f5111b76d2b23063a:

This entire function needs to be marked `unsafe`, or better, you need to tie the lifetime of `bytes` to the `Bitstream` type (perhaps with one of the types in `std::marker`). As written it takes an array `bytes` and then returns an object pointing into `bytes`' data which might outlive `bytes`.

Reply:
I updated `Bitstring` to `Bitstream<'a>` where `'a` is the lifetime of `&bytes`.